### PR TITLE
Add a key fob for the personal vehicle menu.

### PR DESCRIPTION
**Pull request by attributeerror**
When options such as toggle engine, (un)lock doors, toggle lights, etc. are pressed, the player will now face towards the vehicle and press a key fob. It'll play a sound when doing so, much like the GTA:O system.

I attempted to make the key fob object exist but I couldn't be the positioning to be exactly in the right hand. If anyone can do it, they're more than welcome to.

The source & idea for this PR came from [this LSPD:FR script](https://www.lcpdfr.com/files/file/22203-vehicle-keys/).

### CODE REVIEW

**Comment by Doxylamin** [comment]
> I attempted to make the key fob object exist but I couldn't be the positioning to be exactly in the right hand. If anyone can do it, they're more than welcome to.

> The source & idea for this PR came from this LSPD:FR script.

You could take the positioning from the original script, which is defined as below:
```new Model("p_car_keys_01").LoadAndWait();
var keys = new Object("p_car_keys_01", Common.Player.Position);
 if (new Model("p_car_keys_01").IsLoaded)
   keys.AttachTo(Common.Player, Common.Player.GetBoneIndex(PedBoneId.RightHand),
new Vector3(_0.09f, 0.03f, -0.02f_), new Rotator(_-76f, 13f, 28f_));```

File: vMenu/menus/PersonalVehicle.cs [hunk 1]
```diff
@@ -294,6 +299,32 @@ private void CreateMenu()
...
+                if(player.Character.Weapons.Current.Hash != WeaponHash.Unarmed)
+                {
+                    player.Character.Weapons.Give(WeaponHash.Unarmed, 1, true, true);
+                }
+
+                if (!HasEntityClearLosToEntityInFront(player.Character.Handle, CurrentPersonalVehicle.Handle))
+                {
+                    TaskTurnPedToFaceEntity(player.Character.Handle, CurrentPersonalVehicle.Handle, 1000);
+                }
+
```
Comment (Doxylamin) [inline vMenu/menus/PersonalVehicle.cs]:
```suggestion
                uint KeyFobHashKey = (uint) GetHashKey("p_car_keys_01");
                RequestModel(KeyFobHashKey);

                while (!HasModelLoaded(KeyFobHashKey))
                {
                    await Delay(0);
                }

                int KeyFobObject = CreateObject((int) KeyFobHashKey, 0, 0, 0, true, true, true);
                AttachEntityToEntity(KeyFobObject, player.Character.Handle, GetPedBoneIndex(player.Character.Handle, 57005), 0.09f, 0.03f, -0.02f, -76f, 13f, 28f, false, true, true, true, 0, true);
```

File: vMenu/menus/PersonalVehicle.cs [hunk 2]
```diff
@@ -294,6 +299,32 @@ private void CreateMenu()
...
+                RequestAnimDict("anim@mp_player_intmenu@key_fob@");
+                while (!HasAnimDictLoaded("anim@mp_player_intmenu@key_fob@")) {
+                    await Delay(0);
+                }
+                player.Character.Task.PlayAnimation("anim@mp_player_intmenu@key_fob@", "fob_click", 3f, 1000, AnimationFlags.UpperBodyOnly);
+                PlaySoundFromEntity(-1, "Remote_Control_Fob", player.Character.Handle, "PI_Menu_Sounds", true, 0);
```
Comment (Doxylamin) [inline vMenu/menus/PersonalVehicle.cs]:
```suggestion
                PlaySoundFromEntity(-1, "Remote_Control_Fob", player.Character.Handle, "PI_Menu_Sounds", true, 0);

                await Delay(1250);
                DetachEntity(KeyFobObject, false, false);
                DeleteObject(ref KeyFobObject);
```

**Comment by attributeerror** [comment]
I did attempt to use the same code but it didn't work as well as it should've. I'll take a look at the suggested edit tomorrow, thanks!

**Comment by TomGrobbe** [comment]
@attributeerror Let me know whenever you've looked into the key fob object thing, and let me know when it's time to look into merging this. If you're still planning on adding the key fob object then I'll wait for that.

**Pull request merged**